### ISSUE TITLE
NIXIO inconsistent reading and writing of empty ChannelIndex object

### ISSUE THREAD

**Issue by JuliaSprenger** [NeuralEnsemble/python-neo#410]
When saving a ChannelIndex object with an empty index, the object can not be read:

```
import neo
import numpy as np

bl = neo.Block()
bl.channel_indexes.append(neo.ChannelIndex(np.array([]), name='testchx'))
bl.channel_indexes[0].block = bl

filename = 'tmp.nix'
with neo.io.NixIO(filename, 'ow') as nixio:
    nixio.write_block(bl)
with neo.io.NixIO(filename, 'ro') as nixio:
    bl2 = nixio.read_block()
```

results in 

> /python/python-neo/neo/io/nixio.py", line 271, in _source_chx_to_neo
>     if "coordinates" in chx[0]:
> IndexError: list index out of range

**Comment by Christian-B** [comment]
I suggest checking you have neo 0.5.2 (or later) installed.

I had a similar problem with non save channelIndex which I was not able to replicate once I switched to 0.5.2

**Comment by samuelgarcia** [comment]
Is it fix by the new this PR #476 ?

**Comment by achilleas-k** [comment]
Looks like I overlooked this issue. The reader still assumes that there's always at least one channel per ChannelIndex. Fixing now and pushing to open PR.

**Comment by JuliaSprenger** [comment]
This issue will be solved by #476.
